plugins/modules/omevv/omevv_firmware.py

```diff
@@ -0,0 +1,689 @@
+    type: bool
+  drs_check:
+    description:
+      - Allows to check if cluster's DRS is enabled or not.
+      - C(true) checks if cluster's DRS is enabled.
+      - C(false) does not check if cluster's DRS is enabled.
+    type: bool
+    default: false
+  enter_maintenance_mode_options:
+    description:
+      - VM migration policy during management mode.
+      - C(FULL_DATA_MIGRATION) for full data migration.
+      - C(ENSURE_ACCESSIBILITY) for ensuring accessibility.
+      - C(NO_DATA_MIGRATION) does not migrate any data.
+    type: str
+    choices: [FULL_DATA_MIGRATION, ENSURE_ACCESSIBILITY, NO_DATA_MIGRATION]
+  enter_maintenance_mode_timeout:
+    description:
+      - Time out value during maintenance mode in minutes.
+    type: int
+    default: 60
+  evacuate_VMs:
+    description:
+      - Allows to move the VM to other host when host is powered off.
+      - C(true) moves the VM to other host when host is powered off.
+      - C(false) does not move the VM to other host when host is powered off.
+    type: bool
+    default: false
+  exit_maintenance_mode:
+    description:
+      - Whether to exit management mode after Update.
+      - C(true) exits the management mode after Update.
+      - C(false) does not exit the management mode after Update.
+    type: bool
+    default: false
+  job_description:
+    description:
+      - Update job description.
+    type: str
+  job_name:
+    description:
+      - Update job name.
+    type: str
+  job_wait:
+    description:
+      - Whether to wait till completion of the job. This is applicable when I(power_on) is C(true).
+      - C(true) waits for job completion.
+      - C(false) does not wait for job completion.
+    type: bool
+    default: true
+  job_wait_timeout:
+    description:
+      - The maximum wait time of I(job_wait) in seconds. The job is tracked only for this duration.
+      - This option is applicable when I(job_wait) is C(true).
+    type: int
+    default: 1200
+  maintenance_mode_count_check:
+    description:
+      - Allows to check if any host in cluster is in management mode.
+      - C(true) checks if any host in cluster is in management mode.
+      - C(false) does not check if any host in cluster is in management mode.
+    type: bool
+  reboot_options:
+    description:
+      - Host reboot option for firmware update.
+      - C(FORCEREBOOT) will force reboot the server.
+      - C(SAFEREBOOT) will reboot the server in safe mode.
+      - C(NEXTREBOOT) will not reboot the server.
+    type: str
+    choices: [FORCEREBOOT, SAFEREBOOT, NEXTREBOOT]
+    default: SAFEREBOOT
+  reset_idrac:
+    description:
+      - Whether to reset the iDRAC while performing firmware update.
+      - C(true) resets the iDRAC while performing firmware update.
+      - C(false) does not reset the iDRAC while performing firmware update.
+    type: bool
+  run_now:
+    description:
+      - Whether to run the update job now or later.
```

Comment:
•	Whether to run the update job now or later.
•	C(true) runs the update job instantly.
•	C(false) runs the update at the specified I(date_time).

Reply:
Incorporated